src/foundry/common/data/data.mjs/sceneData.d.ts

```diff
@@ -1,23 +1,642 @@
+
+  /**
+   * A darkness level between 0 and 1, beyond which point global illumination is
+   * temporarily disabled if globalLight is true.
+   * @defaultValue `null`
+   */
+  globalLightThreshold?: number | null;
+
+  /**
+   * The ambient darkness level in this Scene, where 0 represents mid-day
+   * (maximum illumination) and 1 represents mid-night (maximum darkness)
+   * @defaultValue `0`
+   */
+  darkness?: number | null;
+
+  /**
+   * A collection of embedded Drawing objects.
+   */
+  drawings?: ConstructorDataType<DrawingData>[] | null;
+
+  /**
+   * A collection of embedded Token objects.
+   */
+  tokens?: ConstructorDataType<TokenData>[] | null;
+
+  /**
+   *
+   * A collection of embedded AmbientLight objects.
+   */
+  lights?: ConstructorDataType<AmbientLightData>[] | null;
+
+  /**
+   * A collection of embedded Note objects.
+   */
+  notes?: ConstructorDataType<NoteData>[] | null;
+
+  /**
+   * A collection of embedded AmbientSound objects.
+   */
+  sounds?: ConstructorDataType<AmbientSoundData>[] | null;
+
+  /**
+   * A collection of embedded MeasuredTemplate objects.
+   */
+  templates?: ConstructorDataType<MeasuredTemplateData>[] | null;
+
+  /**
+   * A collection of embedded Tile objects.
+   */
+  tiles?: ConstructorDataType<TileData>[] | null;
+
+  /**
+   * A collection of embedded Wall objects
+   */
+  walls?: ConstructorDataType<WallData>[] | null;
+
+  /**
+   * A linked Playlist document which should begin automatically playing when this
+   * Scene becomes active.
+   * @defaultValue `null`
+   */
+  playlist?: string | null;
+
+  /**
+   * @defaultValue `null`
+   */
+  playlistSound?: string | null;
+
+  /**
+   * A linked JournalEntry document which provides narrative details about this Scene.
+   * @defaultValue `null`
+   */
+  journal?: string | null;
+
+  /**
+   * A named weather effect which should be rendered in this Scene.
+   * @defaultValue `''`
+   */
+  weather?: string | null;
+
+  /**
+   * The _id of a Folder which contains this Actor
+   * @defaultValue `null`
+   */
+  folder?: string | null;
+
+  /**
+   * The numeric sort value which orders this Actor relative to its siblings
+   * @defaultValue `0`
+   */
+  sort?: number | null;
+
+  /**
+   * An object which configures user permissions to this Actor
+   * @defaultValue `{ default: CONST.ENTITY_PERMISSIONS.NONE }`
+   */
+  permission?: Partial<Record<string, foundry.CONST.EntityPermission>> | null;
+
+  /**
+   * An object of optional key/value flags
+   */
+  flags?: ConfiguredFlags<'Scene'> | null;
 }
 
+/**
+ * The data schema for a Scene document.
+ * @see BaseScene
+ */
 export declare class SceneData extends DocumentData<
-  any,
+  SceneDataSchema,
   SceneDataProperties,
   PropertiesToSource<SceneDataProperties>,
   SceneDataConstructorData,
-  BaseScene
-> {}
+  documents.BaseScene
+> {
+  static defineSchema(): SceneDataSchema;
+
+  /** @override */
+  protected _initialize(): void;
+
+  size: number;
```

Comment:
Any idea where this comes from? It's not part of the schema, is it? foundry just sets
```js
    this.size = Math.round(this.size);
```
in `_initialize`. But that means it already has to be set to _something_...?

Reply:
No, it is not set before, therefore it is initialized to `NaN`.